java/ql/test/library-tests/errortype/Test.java

```diff
@@ -7,6 +7,16 @@ public NoSuchClass test() {
     return nsc;
   }
 
+  static class GenericClass<T> {
+    public void method() { }
+  }
+
+  public testDispatch() {
+    GenericClass<String> g1 = new GenericClass<>();
+    g1.method();
+    GenericClass<NoSuchClass> g2 = new GenericClass<>();
```

Comment:
FWIW, the QL-based type inference/method resolution implementation should be able to handle cases like these.